plugins/maciVoting/contexts/CoordinatorContext.tsx

```diff
@@ -0,0 +1,214 @@
+        };
+      }
+
+      const data = await response.json();
+      return {
+        success: true,
+        data: Boolean(data), // zod is overkill for this
+      };
+    },
+    [getAuthorizationHeader, publicClient]
+  );
+
+  const generateProofs = useCallback(
+    async ({
+      pollId,
+      encryptedCoordinatorPrivateKey,
+      startBlock,
+      endBlock,
+    }: {
+      pollId: number;
+      encryptedCoordinatorPrivateKey: string;
+      startBlock: number;
+      endBlock: number;
+    }): Promise<CoordinatorServiceResult<GenerateResponse>> => {
+      if (!publicClient) {
+        return {
+          success: false,
+          error: new Error("Public client not found"),
+        };
+      }
+
+      const publicKey = await getPublicKey();
+      const encryptedHeader = await getAuthorizationHeader(publicKey);
+
+      const response = await fetch(`${baseUrl}/proof/generate`, {
+        method: "POST",
+        headers: {
+          Authorization: encryptedHeader,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          poll: pollId,
+          maciContractAddress,
+          mode: EMode.NON_QV,
+          encryptedCoordinatorPrivateKey,
+          startBlock,
+          endBlock,
+          blocksPerBatch: 20,
+          chain: publicClient.chain.id,
+        }),
+      }).catch((error) => {
+        throw error instanceof Error ? error : new Error(String(error));
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json();
+        const partialErrorMessage = errorData.message
+          ? `${response.status} - ${response.statusText}. ${errorData.message}`
+          : `${response.status} - ${response.statusText}`;
+
+        return {
+          success: false,
+          error: new Error(`Failed to generate proofs: ${partialErrorMessage}`),
+        };
+      }
+
+      const data = await response.json();
+      return {
+        success: true,
+        data: GenerateResponseSchema.parse(data),
+      };
+    },
+    [getAuthorizationHeader, publicClient]
+  );
+
+  const submit = useCallback(
+    async (pollId: number): Promise<CoordinatorServiceResult<SubmitResponse>> => {
+      if (!publicClient) {
+        return {
+          success: false,
+          error: new Error("Public client not found"),
+        };
+      }
+
+      const publicKey = await getPublicKey();
+      const encryptedHeader = await getAuthorizationHeader(publicKey);
+
+      const response = await fetch(`${baseUrl}/proof/submit`, {
+        method: "POST",
+        headers: {
+          Authorization: encryptedHeader,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          pollId,
+          maciContractAddress,
+          chain: publicClient.chain.id,
+        }),
+      }).catch((error) => {
+        throw error instanceof Error ? error : new Error(String(error));
+      });
```

Comment:
Is there a case where you can have string error instead of `Error`?
If no, you can just remove this catch (and other catches too).

Reply:
I'm not entirely sure but I agree that this could be redundant, removed catch blocks